### PR TITLE
feat(registry): minimal operation chartmuseum etcd

### PR DESCRIPTION
**Pull request by jianzzz**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind feature

**What this PR does / why we need it**:
Chartmuseum supports etcd storage, see https://github.com/helm/chartmuseum#using-with-etcd.
The use of etcd to store chart is supported first, and the minimal run of TKEStack is supported next.
We should replace the version of  storage and etcd, you can see the details in the go.mod file. 
Refer: https://github.com/choujimmy/storage/pull/1

### CODE REVIEW

**Comment by huxiaoliang** [comment]
review comments:
1. please separate this pr only include etcd backend  support, for code format issue, please create a new pr address

**Comment by huxiaoliang** [comment]
1. please separate this pr only include etcd storage support, for code format issue, please create a new pr to address

File: cmd/tke-installer/app/installer/manifests/thanos/thanos.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ---
 apiVersion: v1
 data:
-  thanos.yaml: {{ .ThanosYaml }}
+  thanos.yaml: { { .ThanosYaml } }
```
Comment (leoryu) [inline cmd/tke-installer/app/installer/manifests/thanos/thanos.yaml:4]: Official examples of template use "{{", not "{ {". I'm not sure whether "{{" will be rendered as expected.  I recommend to use what official examples use.

File: cmd/tke-installer/app/installer/manifests/tke-mesh-controller/tke-mesh-controller.yaml [hunk 2]
```diff
@@ -23,15 +23,15 @@ spec:
         nodeAffinity:
           requiredDuringSchedulingIgnoredDuringExecution:
             nodeSelectorTerms:
-            - matchExpressions:
-              - key: node-role.kubernetes.io/control-plane
-                operator: Exists
-            - matchExpressions:
-              - key: node-role.kubernetes.io/master
-                operator: Exists
+              - matchExpressions:
```
Comment (leoryu) [inline cmd/tke-installer/app/installer/manifests/tke-mesh-controller/tke-mesh-controller.yaml:26]: @t1mt @foshantiger please check this change of mesh yaml, thx.

File: cmd/tke-installer/app/installer/manifests/tke-mesh-controller/tke-mesh-controller.yaml [hunk 3]
```diff
@@ -7,7 +7,7 @@ metadata:
   name: tke-mesh-controller
   namespace: tke
 spec:
-  replicas: {{ .Replicas }}
+  replicas: { { .Replicas } }
```
Comment (foshantiger) [inline cmd/tke-installer/app/installer/manifests/tke-mesh-controller/tke-mesh-controller.yaml:10]: use {{  not { {

File: cmd/tke-installer/app/installer/manifests/tke-mesh-controller/tke-mesh-controller.yaml [hunk 4]
```diff
@@ -23,15 +23,15 @@ spec:
...
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/control-plane
+                    operator: Exists
+              - matchExpressions:
+                  - key: node-role.kubernetes.io/master
+                    operator: Exists
       containers:
         - name: tke-mesh-controller
-          image: {{ .Image }}
+          image: { { .Image } }
```
Comment (foshantiger) [inline cmd/tke-installer/app/installer/manifests/tke-mesh-controller/tke-mesh-controller.yaml:34]: use {{ not { {

**Comment by jianzzz** [comment]
> review comments:
> 
> 1. please separate this pr only include etcd storage  support, for code format issue, please create a new pr address
> 2. do we need installer change to make this feature as built-in?
> 3. please add user-guide to doc such as `https://github.com/tkestack/tke/blob/master/docs/design-proposals/appcenter.md`  show we supported  storage back-end, and how to use/switch ?

1. I have recreated a new pr: https://github.com/tkestack/tke/pull/1328
2. We don't need installer change to make this feature as built-in. This is an optional option at this stage.

**Comment by leoryu** [comment]
Close via https://github.com/tkestack/tke/pull/1328

**Pull request closed without merging**